components/wmtk_components/extract_subset/internal/extract_subset_2d.cpp

```diff
@@ -62,23 +62,34 @@ wmtk::TriMesh extract_subset_2d(
 
 
 // Note: the above is a draft version of the algo, implemented in bad, drafted data structure
-// The following is new code to be finished, commented out to compile
+// The following is new code to be finished
 wmtk::TriMesh extract_subset_2d(std::vector<wmtk::Tuple> vertices, std::vector<wmtk::Tuple> triangles, std::vector<size_t> tag){
-//     int nb_vertex = vertices.size();
-//     int nb_vertex_in = 0;
-//     int nb_tri_in = tag.size();
-//     std::vector<bool> vertices_in_bool(nb_vertex);
-//     for (int k = 0; k < nb_vertex; ++k) {vertices_in_bool[k] = false;}
-//     Eigen::MatrixXi faces_in;
-//     faces_in.resize(nb_tri_in, 3);
+    assert(tag.size() <= triangles.size());
 
-//     //tag the preserved ones and count number of vertex in new extraction
-//     for (size_t k = 0; k < nb_tri_in; ++k){
-//         for (size_t k2 = 0; k2 < 3; ++k2) {
-//             // faces_in(k, k2) = triangles(tag[k], k2);
-//             vertices_in_bool[triangles(tag[k], k2)] = true;
-//         }
-//     }
+    int nb_vertex = vertices.size();
+    int nb_vertex_in = 0;
+    int nb_tri_in = tag.size();
+    std::vector<bool> vertices_in_bool(nb_vertex);
+    for (int k = 0; k < nb_vertex; ++k) {vertices_in_bool[k] = false;}
```

Comment:
It looks like you are not using an auto formatter. We have a .clang-format file in our repository. Please install an auto formatter and use this format file. I recommend also to activate "format on save". That way it is sure that you never forget formatting before you commit.